internal/flink/utils.go

```diff
@@ -0,0 +1,52 @@
+package flink
+
+import (
+	"bytes"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/dghubble/sling"
+)
+
+func uploadFile(url, filePath string, formFields map[string]any) error {
+	var buffer bytes.Buffer
+	writer := multipart.NewWriter(&buffer)
+
+	for key, value := range formFields {
+		if strValue, ok := value.(string); ok {
+			_ = writer.WriteField(key, strValue)
+		}
+	}
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	part, err := writer.CreateFormFile("file", filepath.Base(filePath))
+	if err != nil {
+		return err
+	}
+	if _, err := io.Copy(part, file); err != nil {
+		return err
+	}
+
+	if err := writer.Close(); err != nil {
+		return err
+	}
+
+	client := &http.Client{
+		Timeout: 20 * time.Minute,
+	}
+	_, err = sling.New().Client(client).Base(url).Set("Content-Type", writer.FormDataContentType()).Post("").Body(&buffer).ReceiveSuccess(nil)
```

Comment:
Why aren't we using the SDK here?

Reply:
we are using aws presigned url for plugin artifact upload, which requires either aws sdk nor custom connector plugin sdk. this is also clone from https://github.com/confluentinc/cli/blob/3b0527b316868d961060f38b8a47e89f7ecd2a1e/internal/connect/utils.go#L120-L121